xapian-letor/api/featuremanager.cc

```diff
@@ -69,9 +69,9 @@ FeatureManager::getlabel(map<string, map<string, int> > qrel2, const Document &d
 }
 
 Xapian::RankList
-FeatureManager::create_rank_list(const Xapian::MSet & mset, std::string & qid)
+FeatureManager::create_rank_list(const Xapian::MSet & mset, std::string & qid, int TrainorTest)
```

Comment:
It's a check argument for training or testing. While training (TrainorTest == 1), if any doc in mset has no label associated (label == -1), then no featurevector is computed for it.
I'll change the name to lower-case (train_or_test). Even in camelcase it should have been TrainOrTest. Thanks for pointing it out.